airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -442,3 +442,48 @@ def test_send_raise_on_http_errors_logs(mocker, status_code):
         response = stream._send_request(req, {})
         stream.logger.error.assert_called_with(response.text)
         assert response.status_code == status_code
+
+
+@pytest.mark.parametrize(
+    "api_response, expected_message",
+    [
+        ({"error": "something broke"}, "something broke"),
+        ({"error": {"message": "something broke"}}, "something broke"),
+        ({"error": "err-001", "message": "something broke"}, "something broke"),
+        ({"errors": ["one", "two", "three"]}, "one, two, three"),
+        ({"messages": ["one", "two", "three"]}, "one, two, three"),
+        ({"errors": [{"message": "one"}, {"message": "two"}, {"message": "three"}]}, "one, two, three"),
+        ({"errors": [{"error": "one"}, {"error": "two"}, {"error": "three"}]}, "one, two, three"),
+        (["one", "two", "three"], "one, two, three"),
+        ({"error": True}, None),
+        ({"something_else": "hi"}, None),
+        ({}, None),
+    ],
+)
```

Comment:
I love this test-case based development

Reply:
I think a missing case would be:

```
({"error": {"code": "foo-123"}}, "foo-123"),
```

Showing that we prefer to show /something/ from an error object.

What about synonyms for "error"?

```
({"failure": {"message": "foo-123"}}, "foo-123"),
```

Reply:
Added some more cases in [28c355d](https://github.com/airbytehq/airbyte/pull/12944/commits/28c355d5e5080f3c6009cb5853194b54cc777b2e)! I left out `code` for now since I'm not sure how useful it will be for end users to see. We can always add it later though if it does end up being useful.